Add generic ECPoint data class (#73)

iaik-jheher opened the pull request:
rebased version of #73, new PR since i moved it out of the fork to have CI tests

iaik-jheher commented on datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/CryptoPublicKey.kt:
```diff
@@ -327,18 +343,16 @@ sealed class CryptoPublicKey : Asn1Encodable<Asn1Sequence>, Identifiable {
                     byteArrayOf(
-                        prefix,
-                        *x.ensureSize(curve.coordinateLengthBytes)
+                        ANSIECPrefix.forSign(yCompressed).prefixByte,
```
here's a use that would _definitely_ be prettier without `prefixByte`

iaik-jheher reviewed (commented):
@JesusMcCloud re-tagging you in the new PR

JesusMcCloud commented:
@iaik-jheher my last commit is the best i could do, since enums impose some limitations. Fell free to accept as much or as little as you like

iaik-jheher commented:
> @iaik-jheher my last commit is the best i could do, since enums impose some limitations. Fell free to accept as much or as little as you like

lgtm

iaik-jheher commented:
For reference, in addition to the actual EC point stuff, there are two refactors that caused a few LoCs of change:

* Replaced the `Signum = Boolean` typedef with BigInt's `Sign`
* Refactored the `ANSI_*` constants to be an enum class

Since they overlap on quite some lines I'd prefer not to have to split them apart. Sorry for the mess.

JesusMcCloud commented on datatypes/src/commonMain/kotlin/at/asitplus/crypto/datatypes/ECPoint.kt:
please add some kdoc comment explaining what a generalecpoint and an ecpoint is

JesusMcCloud reviewed (approved):
aweseome work except for misssing kdoc as commented. I'm still approving it right away, so you can just rebase once commented.

JesusMcCloud reviewed (approved):
ship it!
